### PR TITLE
Metadata and replay filter syntax

### PR DESCRIPTION
**Pull request by rbdixon**
#### Description

Flows may be filtered based on metadata and replay status.

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/flowfilter.py [hunk 1]
```diff
@@ -382,6 +381,40 @@ def __call__(self, f):
...
+        return f.is_replay == 'response'
+
+
+class FMeta(_Rex):
+    code = "meta"
+    help = "Flow metadata"
+    flags = re.MULTILINE
+
+    def __call__(self, f):
+        m = http.Headers(**{key: repr(value) for key, value in f.metadata.items()})
```
Comment (mhils) [inline mitmproxy/flowfilter.py:414]: That's quite an abuse of our headers class here. Can we search on `f"{key}: {value}"` strings instead please?

**Review by mhils — COMMENTED** [review]
Thanks! This looks good overall. My main point is that I'm not too happy with `~r`, `~rc`, and `~rs`. I feel that the slightly longer `~replay` would be more natural. We should also use `~replayq` and  `~replays`, which is more in line with the other q/s  specialization. Does that make sense? 😃

File: CHANGELOG.md [hunk 2]
```diff
@@ -58,10 +58,9 @@ If you depend on these features, please raise your voice in
 * Add option `export_preserve_original_ip` to force exported command to connect to IP from original request. Only supports curl at the moment. (@dkasak)
 * Major proxy protocol testing (@r00t-)
 * Switch Docker image release to be based on Debian (@PeterDaveHello)
-* Multiple Browsers: The `browser.start` command may be executed more than once to start additional
```
Comment (mhils) [inline CHANGELOG.md:61]: Please re-add, otherwise LGTM. :-)

Comment (rbdixon) [reply CHANGELOG.md:61]: Cleaned up.

**Comment by mhils** [comment]
Excellent, thanks! 🍰

**Pull request merged**